### PR TITLE
Farming: Add support for toolranks mod

### PR DESCRIPTION
**Pull request by louisroyer**
- [toolranks mod](https://github.com/lisacvuk/minetest-toolranks)

### CODE REVIEW

File: mods/farming/api.lua [hunk 1]
```diff
@@ -100,6 +103,15 @@ farming.register_hoe = function(name, def)
 		groups = def.groups,
 		sound = {breaks = "default_tool_breaks"},
 	})
+	-- toolranks optional support
+	if use_toolranks then
+		minetest.override_item(name:sub(2), {
+			original_description = def.description,
+			description = toolranks.create_description(def.description, 0, 1),
+			after_use = toolranks.new_afteruse
```
Comment (sfan5) [inline mods/farming/api.lua]: Why does any of this require support from the farming mod?
I don't see any reason toolranks could not override the item inside it's own mod.

**Comment by SmallJoker** [comment]
How about introducing `farming.registered_hoes`? I'd prefer to export API rather than adding code for specific mods.

**Comment by paramat** [comment]
:-1: We do not base MTG code on specific optional mods, as there are so many we would have to then support everything, and it is impossible to choose which to support.
It is the responsibility of optional mods to alter their code to work with how MTG is.

File: mods/farming/api.lua (hunk 1 above)
Comment (paramat) [reply mods/farming/api.lua]: The toolranks mod should be altered to work with how MTG is, not the other way around.

**Pull request closed without merging**